SMBLibrary/Client/SMB2Client.cs

```diff
@@ -520,7 +530,7 @@ internal SMB2Command WaitForCommand(ulong messageID)
                 }
                 m_incomingQueueEventHandle.WaitOne(100);
             }
-            return null;

```

Comment:
This will change the current flow of returning STATUS_INVALID_SMB. I'm not sold on it and it's a separate change from setting the response timeout, so in any case it shouldn't be in this PR with the other feature

Reply:
It's important to me to differentiate between timeouts and bad requests. I'll split it out though.